pkg/controllers/provisioning/scheduling/scheduler.go

```diff
@@ -205,10 +209,19 @@ func (s *Scheduler) Solve(ctx context.Context, pods []*corev1.Pod) Results {
 	errors := map[*corev1.Pod]error{}
 	QueueDepth.DeletePartialMatch(prometheus.Labels{ControllerLabel: injection.GetControllerName(ctx)}) // Reset the metric for the controller, so we don't keep old ids around
 	q := NewQueue(pods...)
+
+	startTime := s.clock.Now()
+	lastLogTime := s.clock.Now()
+	batchSize := len(q.pods)
 	for {
 		QueueDepth.With(
 			prometheus.Labels{ControllerLabel: injection.GetControllerName(ctx), schedulingIDLabel: string(s.id)},
 		).Set(float64(len(q.pods)))
+
+		if s.clock.Since(lastLogTime) > time.Minute {
+			log.FromContext(ctx).Info("printing scheduling progress", "pods-scheduled", batchSize-len(q.pods), "pods-left", len(q.pods), "duration", s.clock.Since(startTime).Truncate(time.Second))
```

Comment:
Should we use `WithValues` here? I also think we should include scheduling-id to help correlate with metrics.
```suggestion
			log.FromContext(ctx).WithValues("pods-scheduled", batchSize-len(q.pods), "pods-remaining", len(q.pods), "duration", s.clock.Since(startTime).Truncate(time.Second), "scheduling-id", string(s.id)).Info("scheduling progress")
```